File: math/catalan/python/catalan.py
def catalan (n):
	if n <= 1:
		return 1

	num = 1
	for i in range(n + 1, 2 * n + 1):
		num *= i

	den = 1
	for i in range(1, n + 1):
		den *= i

	res = (num / den) / (n + 1)

	return res

File: math/catalan/Java/Catalan.java

public class Catalan {
	public static void main(String [] args) {
		System.out.println(cat(3));
	}

	private static double cat(double m) {
		if(m == 0) return 1 ;
		return (2*m*(2*m-1)/((m+1)*m)) * cat(m-1);
	}
}
